_docs/setup/cluster-extension.md

```diff
@@ -0,0 +1,139 @@
+---
+title: Istio Kubernetes Cluster Extension
+overview: Instructions to add external machines to your Istio managed Kubernetes cluster.
+
+order: 25
+
+layout: docs
+type: markdown
+---
+
+This task shows how to install Istio in a non-kubernetes machine in the same network
+(VPC, VPN) as the Kubernetes cluster.
+
+_This document is under construction._
+
+## Prerequisites
+
+* You have [installed Istio](install-kubernetes.html) on your cluster
+
+* The machine must have IP connectivity to the endpoints in the cluster. This
+typically requires same VPC or a VPN connection, as well as a container network that
+provides direct (without NAT or firewall deny) routing to the endpoints. The machine
+is not required to have access to the cluster IP addresses assigned by K8S.
+
+* The control plane (Pilot, Mixer, CA) and Kubernetes DNS server must be accessible
+from the VM. This is typically done using an [Internal Load
+Balancer](https://kubernetes.io/docs/concepts/services-networking/service/#internal-load-balancer).
```

Comment:
Does this mean that I must have a cloud provider to have this feature worked? Can you please highlight this requirement here?

Reply:
Done. You don't need a cloud provider - you can have on-prem k8s as well, and use node port for example.